x-pack/plugins/security_solution/server/endpoint/services/agent/clients/crowdstrike/crowdstrike_agent_status_client.test.ts

```diff
@@ -0,0 +1,210 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  CrowdstrikeAgentStatusClient,
+  CROWDSTRIKE_NETWORK_STATUS,
+  CROWDSTRIKE_STATUS_RESPONSE,
+} from './crowdstrike_agent_status_client';
+import { NormalizedExternalConnectorClient } from '../../..';
+import { AgentStatusClientError } from '../errors';
+import { HostStatus } from '../../../../../../common/endpoint/types';
+import { CrowdstrikeMock } from '../../../actions/clients/crowdstrike/mocks';
+import { responseActionsClientMock } from '../../../actions/clients/mocks';
+import { savedObjectsClientMock } from '@kbn/core-saved-objects-api-server-mocks';
+import type { SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+
+jest.mock('../../..', () => ({
+  NormalizedExternalConnectorClient: jest.fn(),
+  getPendingActionsSummary: jest.fn().mockResolvedValue([]),
+}));
+
+describe('CrowdstrikeAgentStatusClient', () => {
+  let client: CrowdstrikeAgentStatusClient;
+  const constructorOptions = responseActionsClientMock.createConstructorOptions();
+
+  beforeEach(() => {
+    client = new CrowdstrikeAgentStatusClient({
+      esClient: constructorOptions.esClient,
+      soClient: savedObjectsClientMock.create(),
+      connectorActionsClient: CrowdstrikeMock.createConnectorActionsClient(),
+      endpointService: constructorOptions.endpointService,
+    });
+  });
+
+  describe('getAgentStatusFromConnectorAction', () => {
+    it('should get agent status from connector action', async () => {
+      const agentIds = ['agent1', 'agent2'];
+      const mockExecute = jest.fn().mockResolvedValue({
+        data: {
+          resources: [
+            { id: 'agent1', state: CROWDSTRIKE_STATUS_RESPONSE.ONLINE },
+            { id: 'agent2', state: CROWDSTRIKE_STATUS_RESPONSE.OFFLINE },
+          ],
+        },
+      });
+
+      (NormalizedExternalConnectorClient as jest.Mock).mockImplementation(() => ({
+        setup: jest.fn(),
+        execute: mockExecute,
+      }));
+
+      // @ts-expect-error private method
+      const result = await client.getAgentStatusFromConnectorAction(agentIds);
+
+      expect(mockExecute).toHaveBeenCalledWith({
+        params: {
+          subAction: 'getAgentOnlineStatus',
+          subActionParams: {
+            ids: agentIds,
+          },
+        },
+      });
+
+      expect(result).toEqual({
+        agent1: { id: 'agent1', state: CROWDSTRIKE_STATUS_RESPONSE.ONLINE },
+        agent2: { id: 'agent2', state: CROWDSTRIKE_STATUS_RESPONSE.OFFLINE },
+      });
+    });
+  });
+
+  describe('getAgentStatuses', () => {
+    beforeEach(() => {
+      // @ts-expect-error private method
+      (client.getAgentStatusFromConnectorAction as Jest.Mock) = jest.fn();
+    });
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+    it('should get agent statuses', async () => {
+      const agentIds = ['agent1'];
+      const searchResponse = {
```

Comment:
Being that I think you defined the type incorrectly, can you make sure you add a `type` this variable (and other variables in this PR) so that if you happen to define you mock incorrectly here, at least you will get a type error to signal that it is incorrect.

Reply:
Also - once the mock is fixed to the correct structure - can you ensure you have tests to validate each of the expected values `isolated`, `status` and `found` properties since there is logic that calculates it? 

Here are some ideas:

- Test `isolated` is set correctly for the different values that can appear in the Crowdstrike `status` field
- Test that `found` is correctly set to `false`
- Test the different types of agent `status`'s that the crowdstrike agent can have and validate that we calculate our correct "status" value based on it

Reply:
Thanks Paul, I adjusted the types, and added some tests.